### PR TITLE
`reject` multiple row args support

### PR DESCRIPTION
**Pull request by Tiggax**
# Description
This PR fixes `reject` failing when providing row items in ascending order.

# User-Facing Changes
users can now `reject` multiple rows independently of each other.
```nushell
let foo = [[a b]; [ 1 2] [3 4] [ 5 6]]
# this will work independant of the order
print ($foo | reject 2 1)
print ($foo | reject 1 2)
```

### CODE REVIEW

File: crates/nu-command/tests/commands/reject.rs
Comment (amtoine) [inline crates/nu-command/tests/commands/reject.rs:154]: maybe we can have something exact like the following? :yum: 
```suggestion
    assert_eq!(
        nu!("[[a,b];[1 2] [3 4] [5 6]] | reject 1 2 | to nuon"),
        "[[a, b]; [1, 2]]"
    )
```

File: crates/nu-command/tests/commands/reject.rs
Comment (amtoine) [inline crates/nu-command/tests/commands/reject.rs:164]: maybe we can have something exact like the following? :yum: 
```suggestion
    assert_eq!(
        nu!("[[a,b];[1 2] [3 4] [5 6]] | reject 2 1 | to nuon"),
        "[[a, b]; [1, 2]]"
    )
```

**Review by amtoine — DISMISSED** [review]
> * should `reject`  error when repeated values are passed? Or should it ignore repeats.
>
if we want to not care about the order of indices, maybe repeating them shouldn't matter?
e.g. `reject 1 2`, `reject 2 1` and `reject 1 2 2 1 2` should be the same
> * i think `reject` should error when an index is passed, that is out of bounds ( except for  when optional "?" is present)
>
if it's not a behaviour in either of them, i'd leave that to another PR where one addresses both at once maybe :yum: 

i think these changes are great, i just have suggestions about the tests to test the whole shape of the output data :wink:

**Comment by Tiggax** [comment]
> if we want to not care about the order of indices, maybe repeating them shouldn't matter?
> e.g. `reject 1 2`, `reject 2 1` and `reject 1 2 2 1 2` should be the same

I modeled the function to work similarly to `select` since it also works like that there for rows.
for columns i can add a line to remove repeats, since it fails a test for it right now, so that repeated values wouldn't cause an error.

> if it's not a behavior in either of them, I'd leave that to another PR where one addresses both at once maybe 😋

In the current implementation `reject`  and `select` have operational differences:
|Command|Select|Reject| Works?|
|-|-|-|-|
|pass row that isn't present|empty list|ERROR:access_beyond_end| missmatch?|
|pass column that isn't present|ERROR:column_not_found|ERROR:column_not_found| WAI|
|pass row that isn't present ?|empty list|provided dataframe| WAI|
|pass column that isn't present ?|empty datafame w column name|provided dataframe| WAI |
|repeated row|ERROR:duplicated_row_index|dataframe w.o. row| missmatch?|
|repeated column|dataframe w selected column|dataframe w.o. column| WAI|

For me this is an inconsistency between commands that should work as opposites of each other.
 
I will change `reject` so that it will work even if you pass duplicate columns or rows, since it works like this in the current version.
Then i will maybe open a PR to address these inconsistencies in a different PR.

**Comment by sholderbach** [comment]
Thanks for the side by side table!

**Review by amtoine — DISMISSED** [review]
> I will change `reject` so that it will work even if you pass duplicate columns or rows, since it works like this in the current version.
Then i will maybe open a PR to address these inconsistencies in a different PR.
>
sounds great :pray: 

if i understand correctly, passing the same index multiple times shouldn't error anymore, can we add a test for this behaviour as well? :innocent: 

once that's done, i think we can land this :crossed_fingers:

**Comment by Tiggax** [comment]
@amtoine 
I've added the test for repeated rows. the one for repeated columns is already present in the tests.
This branch should now be good to merge :innocent:

**Comment by fdncred** [comment]
Looks good. Thanks!

**Pull request merged**
